### PR TITLE
Sasssetup

### PR DESCRIPTION
**Pull request by CodesperYu**
Simply created a boilerplate for Sass

I added scss files for components I know we will need styling for: Menu, Chat, Channel (there will be more that I will add)
I added two files called Mixins and Variables
Main.scss file will contain all the other .scss files
App.js will import the Main.scss for styling
Made background-color purple to test scss

### CODE REVIEW

File: src/sass/Components/_Home.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+* {
+	color: white;
```
Comment (miken619) [inline src/sass/Components/_Home.scss]:
```suggestion
  color: white;
```

The outputted CSS file seem to follow the 2 space convention, I think we should keep that same styling for the SCSS files.

File: src/sass/Components/_Home.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+	background-color: purple;
```
Comment (miken619) [inline src/sass/Components/_Home.scss]:
```suggestion
  background-color: purple;
```

File: public/stylesheets/main.css [hunk 3]
```diff
@@ -0,0 +1,5 @@
+* {
+  background-color: purple;
```
Comment (miken619) [inline public/stylesheets/main.css]: Do you know why the outputted file is missing the `color: white;` property?

File: src/sass/Abstracts/_Mixins.scss [hunk 4]
```diff
@@ -0,0 +1 @@
+// Mixins: used for styles that you want to reuse
```
Comment (miken619) [inline src/sass/Abstracts/_Mixins.scss]:
```suggestion
// Mixins: used for styles that you want to reuse.
```

Nit: missing period.

File: src/sass/Abstracts/_Variables.scss [hunk 5]
```diff
@@ -0,0 +1 @@
+// Variables: assign a value to a name for reusability and readability
```
Comment (miken619) [inline src/sass/Abstracts/_Variables.scss]:
```suggestion
// Variables: assign a value to a name for reusability and readability.
```

Nit: missing period.

File: src/sass/Main.scss [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+// run npm start sass:compile to compile to the public/stylesheet folder
+// this file is imported to App.js
+
+//Abstracts
+@import "Abstracts/Mixins";
+@import "Abstracts/Variables";
+
+//Components
+@import "Components/Home";
+@import "Components/Chat";
```
Comment (miken619) [inline src/sass/Main.scss]: I understand the need to set up boilerplate but the `Chat` and `SideMenu` SCSS files don't yet have the associating component setup yet. Can we leave them out for now until we add those components? I just don't want too much 
 noise and empty files.

File: src/sass/Components/_Home.scss (hunk 1 above)
Comment (CodesperYu) [reply src/sass/Components/_Home.scss]: isnt the sass file in src? i named it sass not scss

File: public/stylesheets/main.css (hunk 3 above)
Comment (CodesperYu) [reply public/stylesheets/main.css]: I think it's because i added the white, but i did not run the compiler

File: src/sass/Main.scss (hunk 6 above)
Comment (CodesperYu) [reply src/sass/Main.scss]: Yes, that's totally fine, we can remove them

File: src/sass/Components/_Home.scss (hunk 1 above)
Comment (miken619) [reply src/sass/Components/_Home.scss]: Sorry, my last comment was uncleared. Can we change the name of the directory from `sass` to `scss`? SCSS and SASS uses different syntax and by having a `sass` directory it implies we're following the SASS syntax.

File: src/sass/Abstracts/_Mixins.scss [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<<<<<<< HEAD
```
Comment (miken619) [inline src/sass/Abstracts/_Mixins.scss]: Remove merge comments.

File: src/sass/Abstracts/_Variables.scss [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<<<<<<< HEAD
```
Comment (miken619) [inline src/sass/Abstracts/_Variables.scss]: Remove merge comments.

File: src/sass/Main.scss [hunk 9]
```diff
@@ -0,0 +1,10 @@
+// import all sass files here.
```
Comment (miken619) [inline src/sass/Main.scss]:
```suggestion
// Import all sass files here.
```

File: src/sass/Main.scss [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+// run npm start sass:compile to compile to the public/stylesheet folder.
```
Comment (miken619) [inline src/sass/Main.scss]:
```suggestion
// Run npm start sass:compile to compile to the public/stylesheet folder.
```

File: src/sass/Main.scss [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+// this file is imported to App.js.
```
Comment (miken619) [inline src/sass/Main.scss]:
```suggestion
// This file is imported to App.js.
```

File: src/sass/Main.scss [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+
+//Abstracts
```
Comment (miken619) [inline src/sass/Main.scss]:
```suggestion
// Abstracts.
```

File: src/sass/Main.scss [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+@import "Abstracts/Mixins";
+@import "Abstracts/Variables";
+
+//Components
```
Comment (miken619) [inline src/sass/Main.scss]:
```suggestion
// Components.
```

File: src/sass/Abstracts/_Mixins.scss (hunk 7 above)
Comment (CodesperYu) [reply src/sass/Abstracts/_Mixins.scss]: weird, i dont see this on my end, ill check on why this is appearing

**Comment by miken619** [comment]
Nice!!!

**Pull request merged**